src/components/sideNav.js

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  CDBSidebar,
+  CDBSidebarContent,
+  CDBSidebarFooter,
+  CDBSidebarHeader,
+  CDBSidebarMenu,
+  CDBSidebarMenuItem,
+} from 'cdbreact';
+
+import { NavLink } from 'react-router-dom';
+import { userLogout } from '../store/user/auth';
+import '../App.css';
+
+const Sidenav = () => {
+  const userAuth = localStorage.getItem('userAuth');
+  const dispatch = useDispatch();
+
+  const authState = useSelector((state) => state.authentication);
+
+  const logoutUser = () => {
+    dispatch(userLogout(userAuth));
+  };
+
+  return (
+    <div
+      style={{ display: 'flex', height: '100vh', overflow: 'scroll initial' }}
+    >
+      <CDBSidebar className="sidebar" textColor="#000">
+        <CDBSidebarHeader prefix={<i className="fa fa-bars fa-large" />}>
+          <a
+            href="/"
+            className="text-decoration-none brand"
+            style={{ color: 'inherit' }}
+          >
+            BookaDeV
+          </a>
+        </CDBSidebarHeader>
+
+        <CDBSidebarContent className="sidebar-content">
+          <CDBSidebarMenu>
+            {userAuth || authState.loggedIn ? (
+              <>
+                <NavLink exact to="/">
+                  <CDBSidebarMenuItem icon="user">
+                    DEVELOPERS
+                  </CDBSidebarMenuItem>
+                </NavLink>
+                <NavLink
+                  exact
+                  to="/reservations"
+                >
+                  <CDBSidebarMenuItem icon="calendar">
+                    MY RESERVATIONS
+                  </CDBSidebarMenuItem>
+                </NavLink>
+                {authState.admin ? (
+                  <>
+                    <NavLink
+                      exact
+                      to="/adddeveloper"
+                    >
+                      <CDBSidebarMenuItem icon="plus">
+                        ADD DEVELOPER
+                      </CDBSidebarMenuItem>
+                    </NavLink>
+                    <NavLink
+                      exact
+                      to="/deletedeveloper"
+                    >
+                      <CDBSidebarMenuItem icon="trash">
+                        DELETE DEVELOPER
+                      </CDBSidebarMenuItem>
+                    </NavLink>
+                  </>
+                ) : null}
+                <NavLink exact to="/login">
+                  <CDBSidebarMenuItem
+                    icon="lock"
+                    onClick={logoutUser}
+                    backgroundColor="#f44336"
+                  >
+                    Logout
+                  </CDBSidebarMenuItem>
+                </NavLink>
+              </>
+            ) : (
+              <>
```

Comment:
There ay be a better way to write this code to make it cleaner and neater. A looping method with arrays? try to see if you can make this cleaner [OPTIONAL]

Reply:
Thank @mcrd25